tests/Tests.FeatureManagement/Crontab.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Microsoft.FeatureManagement.FeatureFilters.Crontab;
+using System;
+using Xunit;
+
+namespace Tests.FeatureManagement
+{
+    public class CrontabTest
+    {
+        [Theory]
+        [InlineData("* * * * *", true)]
+        [InlineData("1 2 3 Apr Fri", true)]
+        [InlineData("00-59/3,1,2-2 01,3,20-23 */10,31-1/100 Apr,1-Feb,oct-DEC/1 Sun-Sat/2,1-7", true)]
```

Comment:
Could you explain the `31-1/100` part of the day value? Does cron support the day of the month wrapping around from 31 to 1?

Reply:
The IBM doc and man page of crontab(5) both don't mention whether this is valid.
https://crontab.guru/ this website will consider this invalid. (By the way, this website recognized the step value 100, it only denied range like 100-1)
However, the NCronTab library will treat "31-1/100" as a valid expression and it will be considered as the 31st day of the month.
I think the range like 31-1 and large step value like 100 can be interpreted as the 31st to 1st day of month per 100 days, which will give us only the 31st day of the month.
So I believe this should be valid and supported. This is not a big deal or a critical error.
I can add some extra logics to check whether the step value is within the max value of crontab field and check whether the range is from a smaller number to a a larger number.
What do you think? Should we treat "31-1/100" as a invalid one?

Reply:
I didn't realize NCronTab treated that expression as valid, so it makes more sense to stick with what the NCronTab implementation is and treat that expression as just the 31st day. As long as it's clear that we're sticking with this library then I think it's ok.

Reply:
I think "first-last/step" should be interpreted to something like a for loop: for (int i = first; i<= last; i+=step). If we consider it in this way, everything will make sense.

Reply:
Update: The range like "31-1" should be forbidden. The reason is as follow:
1. The portal should validate the Crontab expression typed by user. If such reversed range is allowed, user may be confused.
2. Although Quartz and NCronTab will treat this as a valid expression, it seems like a bug or a corner case they didn't handle. After more tests, I found that the behavior of NextOccurence(), IsSatisfiedBy(DateTimeOffset) will be extremly strange when both libraries were handling the Crontab like "* * 31-1/100 * *".
For example:
UTs for NCronTab:
```
[InlineData("* * 31-29/2 * *", "Tue, 29 Aug 2023 00:00:00 +00:00", true)]
[InlineData("* * 31-1/30 * *", "Thu, 31 Aug 2023 01:00:00 +00:00", true)]
[InlineData("* * 31-1/30 * *", "Fri, 1 Sep 2023 01:00:00 +00:00", true)]
[InlineData("* * 31-1/31 * *", "Fri, 1 Sep 2023 01:00:00 +00:00", true)]
[InlineData("* * 31-1/31 * *", "Thu, 31 Aug 2023 01:00:00 +00:00", false)]
[InlineData("* * 31-1/100 * *", "Thu, 31 Aug 2023 01:00:00 +00:00", false)]
[InlineData("* * 1-31/100 * *", "Fri, 1 Sep 2023 01:00:00 +00:00", true)]
[InlineData("* * 1-31/100 * *", "Thu, 31 Aug 2023 01:00:00 +00:00", false)]
public void IsCronSatisfiedByTimestamp(string cron, string timeString, bool expected)
{
    Assert.Equal(expected, IsSatisfiedBy(cron, timeString));
}
private static bool IsSatisfiedBy(string cron, string timeString)
{
    CrontabSchedule schedule = CrontabSchedule.Parse(cron, new CrontabSchedule.ParseOptions { IncludingSeconds = false });

    DateTime dateTime = DateTime.Parse(timeString);
    DateTime withoutSeconds = new DateTime(dateTime.Year, dateTime.Month, dateTime.Day, dateTime.Hour, dateTime.Minute, 0);
    DateTime timeAfter = schedule.GetNextOccurrence(withoutSeconds.AddMinutes(-1));
    if (timeAfter.Equals(withoutSeconds))
    {
        return true;
    } 
    else
    {
        return false;
    } 
}
```